### PR TITLE
Prettify the TEG

### PR DESCRIPTION
**Pull request by martinlyra**
## About The Pull Request

DISCLAIMER: The original author is @retlaw34 and I take no credit for this

- Ports the new aesthetics from https://github.com/yogstation13/Yogstation/pull/9195
- Additionally, we (me and retlaw34) have made a preliminary tgUI implementation of the UI
- ~~Currently, this PR also adds a tiny debug ship for testing the TEG.~~

## Why It's Good For The Game

Pretty TEG icons.

## Changelog
:cl: martinlyra & retlaw34
tweak: Changed the TEG's UI to tgUI
imageadd: The TEG's visuals have been completely redone. Sprites by Pawn.
/:cl:

### CODE REVIEW

**Comment by martinlyra** [comment]
Two slightly old previews because I am a lazy tic-tac-toe

image
image

**Comment by MarkSuckerberg** [comment]
@triplezeta

**Review by triplezeta — APPROVED** [review]
sprites okgood

File: code/modules/power/generator.dm [hunk 1]
```diff
@@ -140,17 +150,30 @@
...
+	if(!ui)
+		ui = new(user, src, "ThermalGenerator", name)
+		ui.open()
+		ui.set_autoupdate(TRUE)
 
+/obj/machinery/power/generator/ui_data(mob/user)
+//	var/datum/gas_mixture/cold_circ_air1 = cold_circ.airs[1] //outlet cold
+	var/datum/gas_mixture/cold_circ_air2 = cold_circ.airs[2] //inlet cold
+//	var/datum/gas_mixture/hot_circ_air1 = hot_circ.airs[1] //outlet hot
+	var/datum/gas_mixture/hot_circ_air2 = hot_circ.airs[2] //inlet hot
```
Comment (MarkSuckerberg) [inline code/modules/power/generator.dm:122]: Any reason for the commented out vars?

File: code/modules/power/generator.dm [hunk 2]
```diff
@@ -99,6 +108,7 @@
 	lastgen -= power_output
 	..()
 
+/* todo: destroy
```
Comment (MarkSuckerberg) [inline code/modules/power/generator.dm]: destroy

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/modules/power/generator.dm (hunk 1 above)
Comment (martinlyra) [reply code/modules/power/generator.dm:122]: For future work. They each hold the pressures and temperatures of *before* and *after* for each circulator, but for now we are just sending the *before* data.

File: code/modules/power/generator.dm (hunk 2 above)
Comment (ZephyrTFA) [reply code/modules/power/generator.dm]: @martinlyra

Comment (rye-rice) [reply code/modules/power/generator.dm]: Right so, this was only comented out when I needed to glance at how the original HTML ui worked, you dont need this anymore

**Comment by martinlyra** [comment]
Pardon me, I was in middle of university homework and running a play-test. I should be able to get onto the issues of this PR now.

File: code/modules/power/generator.dm (hunk 2 above)
Comment (martinlyra) [reply code/modules/power/generator.dm]: destroyed:tm:

File: code/modules/power/generator.dm [hunk 3]
```diff
@@ -159,70 +138,120 @@
...
+		if(C && C.dir == SOUTH && C.anchored && !(C.machine_stat &(BROKEN)) && !C.panel_open)
 			circs += C
 
 		C = locate(circpath) in get_step(src, SOUTH)
-		if(C && C.dir == NORTH)
+		if(C && C.dir == NORTH && C.anchored && !(C.machine_stat &(BROKEN)) && !C.panel_open)
 			circs += C
 
-	if(circs.len)
+	if(circs.len == 2)
```
Comment (MarkSuckerberg) [inline code/modules/power/generator.dm]:
```suggestion
	if(length(circs) == 2)
```
I have become "that" level of maintainer, yes.

File: code/modules/power/generator.dm [hunk 4]
```diff
@@ -159,70 +138,120 @@
...
 			if(C.mode == CIRCULATOR_COLD && !cold_circ)
 				cold_circ = C
 				C.generator = src
 			else if(C.mode == CIRCULATOR_HOT && !hot_circ)
 				hot_circ = C
 				C.generator = src
+		if(!hot_circ || !cold_circ)
+			kill_circs()
+			return 3
+	return circs.len
```
Comment (MarkSuckerberg) [inline code/modules/power/generator.dm]:
```suggestion
	return length(circs)
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
I requested these changes like three or four days ago but forgot to finish the review, whoooops

**Review by MarkSuckerberg — APPROVED** [review]

**Pull request merged**